### PR TITLE
[Frontend] Pass capabilities to the decomposer

### PR DESCRIPTION
**Pull request by grwlf**
**Context:** Catalyst needs to deal with operations which are not supported by the target device. This PR addresses decomposition-to-matrix branch. 

**Description of the Change:** We read gates to be decomposed to matrix from the device capabilities (which in turn are described in the device toml file).
[sc-62011]

**Benefits:** Device authors gain control over the decomposition strategies

**Possible Drawbacks:**

**Related GitHub Issues:**
Requires https://github.com/PennyLaneAI/catalyst/pull/712

### CODE REVIEW

File: doc/changelog.md [hunk 1]
```diff
@@ -82,9 +82,14 @@
...
 
 <h3>Bug fixes</h3>
+
 * Correctly querying batching rules for `jax.scipy.linalg.expm`
   [(#733)](https://github.com/PennyLaneAI/catalyst/pull/733)
 
+* Correctly linking openblas routines necessary for `jax.scipy.linalg.expm`.
+  In this bug fix, four openblas routines were newly linked and are now discoverable by `stablehlo.custom_call@<blas_routine>`. They are `blas_dtrsm`, `blas_ztrsm`, `lapack_dgetrf`, `lapack_zgetrf`.
+  [(#752)](https://github.com/PennyLaneAI/catalyst/pull/752)    
+
```
Comment (lillian542) [inline doc/changelog.md:92]: This entry is for a different PR

File: frontend/test/pytest/test_preprocess.py [hunk 2]
```diff
@@ -582,7 +628,7 @@ def test_controlled_decomposes_to_unitary(self):
         tape = qml.tape.QuantumScript([qml.ctrl(qml.RX(1.23, 0), 1)])
 
         with EvaluationContext(EvaluationMode.QUANTUM_COMPILATION) as ctx:
-            (new_tape,), _ = catalyst_decompose(tape, ctx, stopping_condition)
+            (new_tape,), _ = catalyst_decompose(tape, ctx, stopping_condition, capabilities)
```
Comment (lillian542) [inline frontend/test/pytest/test_preprocess.py:632]: Can we add a test similar to this test that checks the behaviour added in this PR, i.e. that whether or not an operator is in the list of ops to decompose to unitary determined whether we decompose it or not?

**Review by lillian542 — COMMENTED** [review]
Looks good! Left a few minor comments.

File: frontend/test/pytest/test_jax_dynamic_api.py [hunk 3]
```diff
@@ -400,33 +398,5 @@ def fun(sz, idx, val):
...
-        return jsp.linalg.expm(x)
-
-    y2 = jnp.array([[1.0, 0.0], [0.0, 1.0]])
-    res2 = f2(y2)
-    expected2 = jnp.array([[2.71828183, 0.0], [0.0, 2.71828183]])
-
-    assert_allclose(res1, expected1)
-    assert_allclose(res2, expected2)
-
-
```
[the remark this answers is not in the document]
Comment (josh146) [reply frontend/test/pytest/test_jax_dynamic_api.py:430]: @lillian542 it seems like this PR has accidentally pulled in changes for https://github.com/PennyLaneAI/catalyst/pull/752

File: doc/changelog.md (hunk 1 above)
Comment (grwlf) [reply doc/changelog.md]: This seems to be a Github bug. I changed the pr's base to main, it is gone.

File: frontend/catalyst/utils/libcustom_calls.cpp [hunk 4]
```diff
@@ -13,6 +13,7 @@
 // limitations under the License.
 
 #include <cmath>
```
[the remark this answers is not in the document]
Comment (grwlf) [reply frontend/catalyst/utils/libcustom_calls.cpp]: A github bug probably. Changed base to main.

File: frontend/test/pytest/test_jax_dynamic_api.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (grwlf) [reply frontend/test/pytest/test_jax_dynamic_api.py]: I can't imagine how it happens. Changed pr's base to main.

File: frontend/test/pytest/test_jax_numerical.py [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Test that numerical jax functions produce correct results when compiled with qml.qjit"""
+
+import numpy as np
```
[the remark this answers is not in the document]
Comment (grwlf) [reply frontend/test/pytest/test_jax_numerical.py]: Changed pr's base to main.

File: frontend/test/pytest/test_preprocess.py (hunk 2 above)
Comment (grwlf) [reply frontend/test/pytest/test_preprocess.py:649]: [Done](https://github.com/PennyLaneAI/catalyst/blob/3448527ab3499c984dc782d091ece69978b58789/frontend/test/pytest/test_preprocess.py#L626)

**Pull request merged**